serving/samples/helloworld-dart/README.md

```diff
@@ -78,7 +78,6 @@ be created using the following instructions.
 
    # Configure and document the service HTTP port.
    ENV PORT 8080
```

Comment:
You should remove PORT, this is injected by Knative.

Reply:
(true for all samples)

Reply:
I prefer to keep the explicit default as a way of documenting the contract and making it obvious how to override the port for experimentation.

Reply:
8080 is not part of the runtime contract, only PORT is: https://github.com/knative/serving/blob/master/docs/runtime-contract.md#process

Reply:
Explicitly showing this remains valuable, even if the value is a common convention rather than an official default.

Reply:
Explicitly showing this remains valuable, even if the value is a common convention rather than an official default.

Reply:
Just to make sure we have the same definition here, the ENV directive is a way to set a default environment variable value at build time which is overridden at run time by any mechanism which injects environment variables into the container.

I see including this line as a way of:

1. Having sensible defaults to ensure the container will work locally.
2. Documenting what the PORT env var is about for practitioners that read the Dockerfile but not the docs.

Reply:
My concerns is that users will assume these are the default values of `PORT`.

As far as I understand, the `ENV` directive is not needed for running locally when using `docker -p`.

Running locally is better done using `$ PORT=8080 docker run -p 8080:${PORT} -e PORT=${PORT} [IMAGE]`

Reply:
This is setting a default value for the service, which is a good practice. It also allows the container to be spun up with a more typical and abbreviated command:

```
docker run -p 8080 [IMAGE]
```

A good amount of the value in this is if developers take the Dockerfile and pivot to using this as part of an ongoing service.

The Day 1 value is in surfacing a bit of the runtime contract for developers that look at code first. What if a comment were added indicating that 8080 is not a requirement but a sane default?

Reply:
OK to leave `ENV PORT 8080` in the Dockerfiles, but indeed please change the comment, "Configure the service HTTP port." is not correct. The service HTTP port is not configured via this env var. Use something like "Service must listen on $PORT, set value for local development"